### PR TITLE
Rough solution to buyers guide supported locales.

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

**Comment by ScottDowne** [comment]
@TheoChevalier this seems to handle it on my end.

I did end up getting the strings to separate with multiple json files.

File: scripts/properties2json.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-var config = require('../intl-config.js');
+var defaultConfig = require('../intl-config.js');
```
Comment (ScottDowne) [inline scripts/properties2json.js]: All of the changes in this file have one goal.

Making things generic enough so we can plug in different configs.

File: scripts/properties2json.js [hunk 2]
```diff
@@ -62,30 +71,37 @@ function readPropertiesFile(filePath) {
...
-getListLocales().then(processMessageFiles)
-.then(writeFile).catch(function(err) {
-  console.error(err);
+applyConfig(defaultConfig);
+applyConfig({
+  "dest": {
+    "dir": "public",
+    "file": "buyers-guide-locales.json"
+  },
+  "src": "buyers-guide-locales"
```
Comment (ScottDowne) [inline scripts/properties2json.js]: The above few lines are the main point. Anything in this file that was like, config.something we a problem, now it's not.

File: src/client.js [hunk 3]
```diff
@@ -6,13 +6,13 @@ import { Router, browserHistory } from 'react-router';
 import { IntlProvider, addLocaleData } from 'react-intl';
 import routes from './routes.js';
 import ga from 'react-ga';
-import locales from '../public/locales.json';
 import assign from 'object-assign';
+import getMessages from './lib/get-messages.js';
```
Comment (ScottDowne) [inline src/client.js]: getMessages just returns the set of strings based on the path you pass it.

File: src/lib/get-messages.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import defaultLocales from '../../public/locales.json';
```
Comment (ScottDowne) [inline src/lib/get-messages.js]: These are just stubs for now. I have them more fleshed out in the other patch but it made sense to separate them.

File: src/pages/buyers-guide/category.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+var React = require(`react`);
```
Comment (ScottDowne) [inline src/pages/buyers-guide/category.js]: These are just stubs for now. I have them more fleshed out in the other patch but it made sense to separate them.

File: src/routes.js [hunk 6]
```diff
@@ -107,6 +107,22 @@ function buildRoutes() {
       </Route>
     );
   });
+
+  ["en-US", "es"].forEach(function(locale) {
```
Comment (ScottDowne) [inline src/routes.js]: I didn't think this would work as well as it did.

Basically I was able create a new set of locales just for the new routes.

Comment (ScottDowne) [reply src/routes.js]: so /de/buyers-guide doesn't exist.

Comment (TheoChevalier) [reply src/routes.js]: Nice, so what will I see with a German browser, it will redirect to /en-US/buyers-guide?

Comment (ScottDowne) [reply src/routes.js]: That is correct 🚀

Comment (ScottDowne) [reply src/routes.js]: or es if that's your locale set in the browser.

**Comment by TheoChevalier** [comment]
Overall, it seems to cover our needs, thanks a lot Scott!

Just checking a small detail with Matjaz, as I actually may not need a separate repository.

**Pull request merged**